public/app/plugins/panel/heatmap-new/utils.test.ts

```
@@ -0,0 +1,330 @@
+    });
+
+    it('filters out minimum and maximum values', () => {
+      expect(mapping.flat()).not.toContainEqual(1);
+      expect(mapping.flat()).not.toContainEqual(10);
+    });
+  });
+
+  describe('Error scenarios', () => {
+    const heatmap = {
+      heatmap: {
+        name: 'test',
+        meta: {
+          type: DataFrameType.HeatmapBuckets,
+        },
+        fields: [
+          {
+            name: 'xMin',
+            type: FieldType.number,
+            config: {},
+            values: new ArrayVector([1]),
+          },
+          {
+            name: 'yMin',
+            type: FieldType.number,
+            config: {},
+            values: new ArrayVector([1, 4, 7]),
+          },
+          {
+            name: 'count',
+            type: FieldType.number,
+            config: {},
+            values: new ArrayVector([2, 0, 6]),
+          },
+        ],
+        length: 3,
+      },
+      xBucketCount: 1,
+      xBucketSize: 9,
+      xLayout: BucketLayout.ge,
+      yBucketCount: 3,
+      yBucketSize: 3,
+      yLayout: BucketLayout.ge,
+    };
+    const origData = {
+      name: 'origdata',
+      fields: [
+        {
+          name: 'time',
+          type: FieldType.time,
+          config: {},
+          values: new ArrayVector([1, 2, 3, 4, 5, 6, 7, 8, 9]),
+        },
+        {
+          name: 'value',
+          type: FieldType.number,
+          config: {},
+          values: new ArrayVector([7, 1, 8, 7, 2, 7, 8, 8, 4]),
+        },
+      ],
+      length: 2,
+    };
+
+    it('Will not process heatmap buckets', () => {
+      expect(
+        getDataMapping(
+          {
+            ...heatmap,
+            heatmap: {
+              ...heatmap.heatmap,
+              meta: {
+                type: DataFrameType.HeatmapBuckets,
+              },
+            },
+          },
+          origData
+        )
+      ).toEqual([null]);
```

should we throw an error?  it feels more like something we should catch loudly than just return empty values

---

I can definitely throw an error.